File: lagom/examples/hello-to-greeting/hello-to-greeting-impl/src/main/scala/com/apuex/greeting/mapping/h2g/H2GMappingServiceImpl.java
package com.apuex.greeting.mapping.h2g;


import akka.NotUsed;
import akka.stream.javadsl.Source;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import javax.inject.Inject;
import java.util.Date;
import java.util.Optional;

import static java.util.concurrent.CompletableFuture.completedFuture;

class H2GMappingServiceImpl implements H2GMappingService {
  @Inject
  private MappingConfig config;

  /**
   * Subscribe from event stream with offset.
   *
   * @param offset timed-uuid or sequence specifies start position
   * @return
   */
  @Override
  public ServiceCall<Source<String, NotUsed>, Source<String, NotUsed>> events(Optional<String> offset) {
    System.out.println(offset);
    return inputs -> completedFuture(
        inputs.map(x -> String.format("[%s] - %s", new Date(), x))
    );
  }

}

File: lagom/examples/hello-to-greeting/hello-to-greeting-impl/src/main/scala/com/apuex/greeting/mapping/h2g/H2GMappingServiceImpl.scala
package com.apuex.greeting.mapping.h2g

import akka.NotUsed
import akka.stream.scaladsl.Source
import com.lightbend.lagom.scaladsl.api._

import scala.concurrent.Future

class H2GMappingServiceImpl(config: MappingConfig) extends H2GMappingService {

  /**
    * Subscribe from event stream with offset.
    *
    * @param offset timed-uuid or sequence specifies start position
    * @return
    */
  def events(offset: Option[String]): ServiceCall[Source[String, NotUsed], Source[String, NotUsed]] = ServiceCall { is =>
    Future.successful(is.map(x => x))
  }
}
